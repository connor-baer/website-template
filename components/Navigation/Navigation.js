import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import { hideVisually } from 'polished';

import CONFIG from '../../config';
import Link from '../Link';
import Logo from './logo.svg';
import { fullWidthStyles } from '../../styles/shared';

const { site } = CONFIG;

const wrapperStyles = ({ theme }) => css`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;

  ${theme.mq.kilo} {
    padding-top: 2rem;
    padding-bottom: 2rem;
  }
`;

const Wrapper = styled('div')(wrapperStyles, fullWidthStyles);

const logoStyles = ({ theme }) => css`
  fill: ${theme.colors.p500};
  width: 2.5rem;
  height: 2.5rem;
`;

const StyledLogo = styled(Logo)(logoStyles);

const SiteName = styled('span')`
  ${hideVisually()};
`;

const linkStyles = ({ theme }) => css`
  margin-left: 1rem;
  margin-right: -1rem;
  padding: 1rem;
  color: inherit;
  text-decoration: none;
  transition: color 0.1s ease-in-out, border 0.1s ease-in-out;

  &:hover,
  &:focus {
    color: ${theme.colors.p500};
    text-decoration: underline;
  }

  &:focus {
    outline: thin dotted currentColor;
    outline-offset: 0.25em;
  }

  &:active {
    color: ${theme.colors.p500};
  }

  ${theme.mq.kilo} {
    margin-left: 2rem;
  }

  ${theme.mq.mega} {
    margin-left: 3rem;
  }
`;

const A = styled('a')(linkStyles);

export default function Navigation({
  siteName = site.name,
  links = site.navLinks,
  ...rest
}) {
  return (
    <Wrapper>
      <Link href="/">
        <a title="Home">
          <StyledLogo aria-labelledby="site-name" />
          <SiteName id="site-name">{siteName}</SiteName>
        </a>
      </Link>

      <nav {...rest}>
        {links.map(({ label, ...link }) => (
          <Link key={link.href} {...link} passHref>
            <A>{label}</A>
          </Link>
        ))}
      </nav>
    </Wrapper>
  );
}
