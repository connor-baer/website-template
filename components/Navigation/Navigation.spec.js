import React from 'react';

import Navigation from './Navigation';

describe('Navigation', () => {
  function renderNavigation(fn, props = {}) {
    return fn(<Navigation {...props} />);
  }

  it('should render with default styles', () => {
    const { container } = renderNavigation(render);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const html = renderNavigation(renderToHtml);
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
