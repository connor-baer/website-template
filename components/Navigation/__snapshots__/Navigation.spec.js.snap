// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render with default styles 1`] = `
.circuit-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
  max-width: 80rem;
  margin-right: auto;
  margin-left: auto;
}

@media (min-width:600px) {
  .circuit-4 {
    padding-top: 2rem;
    padding-bottom: 2rem;
  }
}

@media (min-width:600px) {
  .circuit-4 {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}

.circuit-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.circuit-2 {
  margin-left: 1rem;
  margin-right: -1rem;
  padding: 1rem;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.1s ease-in-out,border 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,border 0.1s ease-in-out;
}

.circuit-2:hover,
.circuit-2:focus {
  color: #f06595;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.circuit-2:focus {
  outline: thin dotted currentColor;
  outline-offset: 0.25em;
}

.circuit-2:active {
  color: #f06595;
}

@media (min-width:600px) {
  .circuit-2 {
    margin-left: 2rem;
  }
}

@media (min-width:900px) {
  .circuit-2 {
    margin-left: 3rem;
  }
}

<div
  class="circuit-4 circuit-5"
>
  <a
    href="/"
    title="Home"
  >
    <div>
      logo.svg
    </div>
    <span
      class="circuit-0 circuit-1"
      id="site-name"
    >
      Website Template
    </span>
  </a>
  <nav>
    <a
      class="circuit-2 circuit-3"
      href="/about"
    >
      About
    </a>
  </nav>
</div>
`;
